src/components/GlobalCheckClaimStatus/index.tsx

```diff
@@ -29,33 +32,55 @@ const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = (props) =>
  */
 const GlobalCheckClaim: React.FC<GlobalCheckClaimStatusProps> = ({ excludeLocations }) => {
   const hasDisplayedModal = useRef(false)
-  const [canClaimAnniversaryPoints, setCanClaimAnniversaryPoints] = useState(false)
+  const [cid, setCid] = useState(null)
+  const [canClaimNFT, setCanClaimNFT] = useState(false)
+  const [nftBalance, setNftBalance] = useState(999) // default high to avoid fashing modal
+  const galaxyNFTContract = useERC721('0x4344cDD59780CC7B8A47Cd903424F91f41A3a366')
   const { account } = useWeb3React()
   const { pathname } = useRouter()
-  const [onPresentAnniversaryModal] = useModal(<AnniversaryAchievementModal />)
+  const [onPresentModal] = useModal(<GalaxyNFTClaimModal cid={cid} />, false, true, 'galaxyNFTClaimModal')
 
   // Check claim status
   useEffect(() => {
-    const fetchClaimAnniversaryStatus = async () => {
-      const { canClaim } = getAnniversaryAchievementContract()
-      const canClaimAnniversary = await canClaim(account)
-      setCanClaimAnniversaryPoints(canClaimAnniversary)
+    const fetchClaimNftStatus = async () => {
+      try {
+        const { campaign } = await request(
```

Comment:
We can put these two async calls in Promise.all

Reply:
nah, let's just release it, its not gonna be live for long time anyway 😂